pkg/arvo/sys/vane/khan.hoon

```diff
@@ -0,0 +1,148 @@
+::  %khan, thread runner
+!:
+!?  164
+::
+|=  our=ship
+=>  |%
+    +$  move  [p=duct q=(wite note gift:khan)]
+    +$  note                                            ::  out request $->
+      $~  [%g %deal *sock *term *deal:gall]
+      $%  $:  %g                                        ::  to %gall
+              $>(%deal task:gall)                       ::  full transmission
+      ==  ==                                            ::
+    +$  sign                                            ::  in response $<-
+      $%  $:  %gall
+              $>(%unto gift:gall)
+          ==
+          $:  %khan
+              $>(%avow gift:khan)
+      ==  ==
+    +$  khan-state
+      $:  %0
+          hey=duct                                      ::  current unix duct
+          run=(map duct thread-state)                   ::  running threads
+      ==
+    ::  TODO  move to lull
+    ::
+    +$  thread-state  [tid=@ta mak=mark bek=beak]
+    --
+=>
+|%
+++  get-dais
+  |=  [=beak =mark rof=roof]
+  ^-  dais:clay
+  ?~  ret=(rof ~ %cb beak /[mark])
+    ~|(%mark-unknown !!)
+  ?~  u.ret
+    ~|(%mark-invalid !!)
+  ?>  =(%dais p.u.u.ret)
+  !<(dais:clay q.u.u.ret)
+++  start-spider
+  |=  [our=@p =cage]
+  ^-  note
+  [%g %deal [our our] %spider %poke cage]
```

Comment:
@philipcmonk should either end of the sock change if we request a thread from an external `+beak`?

Reply:
@Fang- says `[our our]` is correct here